src/persist-client/src/internal/trace.rs

```diff
@@ -1027,7 +1027,7 @@ impl<T: Timestamp + Lattice> Spine<T> {
     /// could do so in order to maintain fewer batches on average (at the risk
     /// of completing merges of large batches later, but tbh probably not much
     /// later).
-    pub fn apply_fuel(&mut self, fuel: &mut isize, log: &mut SpineLog<'_, T>) {
+    pub fn apply_fuel(&mut self, fuel: &isize, log: &mut SpineLog<'_, T>) {
```

Comment:
@danhhz might want to look at this, it looks like a bug?

Reply:
This is forked from differential dataflow. Persist had made a few modifications, but I just checked and it doesn't look like we changed this bit. My guess is that this is a result of this comment

```
        // For the moment our strategy is to apply fuel independently to each
        // merge in progress, rather than prioritizing small merges. This sounds
        // like a great idea, but we need better accounting in place to ensure
        // that merges that borrow against later layers but then complete still
        // "acquire" fuel to pay back their debts.
```